health-services/MeasureData/app/src/main/java/com/example/measuredata/HealthServicesManager.kt

```diff
@@ -49,31 +48,35 @@ class HealthServicesManager @Inject constructor(
      *
      * [callbackFlow] is used to bridge between a callback-based API and Kotlin flows.
      */
-    fun heartRateMeasureFlow() = callbackFlow<MeasureMessage> {
+    @ExperimentalCoroutinesApi
+    fun heartRateMeasureFlow() = callbackFlow {
         val callback = object : MeasureCallback {
-            override fun onAvailabilityChanged(dataType: DataType, availability: Availability) {
+            override fun onAvailabilityChanged(dataType: DeltaDataType<*, *>, availability: Availability) {
                 // Only send back DataTypeAvailability (not LocationAvailability)
                 if (availability is DataTypeAvailability) {
-                    sendBlocking(MeasureMessage.MeasureAvailabilty(availability))
+                    trySendBlocking(MeasureMessage.MeasureAvailability(availability))
                 }
             }
 
-            override fun onData(data: List<DataPoint>) {
-                sendBlocking(MeasureMessage.MeasureData(data))
+            override fun onDataReceived(data: DataPointContainer) {
+                val heartRateBpm = data.getData(DataType.HEART_RATE_BPM)
+                trySendBlocking(MeasureMessage.MeasureData(heartRateBpm))
             }
         }
 
         Log.d(TAG, "Registering for data")
-        measureClient.registerCallback(DataType.HEART_RATE_BPM, callback)
+        measureClient.registerMeasureCallback(DataType.HEART_RATE_BPM, callback)
 
```

Comment:
Do you think we should include an example of when to handle `onRegistrationFailed` ? I am not sure of whether there are scenarios we'd expect the developer to check here, or in what scenarios a failure can occur?